<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>2024_02_09_Code_Review</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="2024_02_09_Code_Review_files/libs/clipboard/clipboard.min.js"></script>
<script src="2024_02_09_Code_Review_files/libs/quarto-html/quarto.js"></script>
<script src="2024_02_09_Code_Review_files/libs/quarto-html/popper.min.js"></script>
<script src="2024_02_09_Code_Review_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="2024_02_09_Code_Review_files/libs/quarto-html/anchor.min.js"></script>
<link href="2024_02_09_Code_Review_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="2024_02_09_Code_Review_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="2024_02_09_Code_Review_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="2024_02_09_Code_Review_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="2024_02_09_Code_Review_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-full">

<main class="content column-page" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">2024_02_09_Code_Review</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="experimental-design" class="level1">
<h1>Experimental Design</h1>
<p>We are researching the use of Ruxolitnib, a JAK1/2 kinase inhibitor, to suppress the immune system during bone marrow transplants (BMT) to increase donor engraftment.</p>
<p>In this experiment, we collected bulk RNAseq data of lymphocytes isolated from the spleens of mice who received bone marrow transplants under varying conditions.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Code%20Review%20Figures/BMT%20Rux%20RNAseq%20Experimental%20Model.png" class="img-fluid figure-img"></p>
</figure>
</div>
</section>
<section id="experiment-conditions" class="level1">
<h1>Experiment Conditions</h1>
<p>We used two strains of mice in this experiment:</p>
<ul>
<li><p>C57BL/6 AKA <code>Black 6</code></p></li>
<li><p><code>Balb/c</code></p></li>
</ul>
<p>These mice have slightly different forms of the mouse MHC I (AKA HLA) protein, called H2Kb and H2Kd, respectively.</p>
<p>BMTs that use the same mouse strain as donor and recipient are considered autologous.</p>
<p>BMTs that use different strains for donor and recipient are considered allogenic, and these simulate conditions BMT recipients face in the hospital.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Code%20Review%20Figures/Mouse%20Design.png" class="img-fluid figure-img" width="900"></p>
</figure>
</div>
<p>Using these conditions, we hope to identify genes involved in transplant engraftment (acceptance), rejection, and tolerance of allogenic bone marrow cells.</p>
</section>
<section id="load-libraries" class="level1">
<h1>Load Libraries</h1>
<p>These libraries are required.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>({</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pcaExplorer)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Mm.eg.db)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(topGO)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GeneTonic)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EnhancedVolcano)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggplot2' was built under R version 4.3.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
groupGOTerms:   GOBPTerm, GOMFTerm, GOCCTerm environments built.</code></pre>
</div>
</div>
<p>Increasing available memory.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">future.globals.maxSize =</span> <span class="dv">14000</span> <span class="sc">*</span> <span class="dv">1024</span><span class="sc">^</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read-in-metadata" class="level1">
<h1>Read in metadata</h1>
<p>Let’s read in the metadata and look at the different conditions. The NK cell samples have a n=1 and we won’t use them now.</p>
<p>We also found some data online for naive Black 6 CD4 and CD8 T cells which I included under the name “WT Control.”</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">"2023_11_09_results/combined_meta.csv"</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>meta <span class="sc">|&gt;</span> </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(cell_type <span class="sc">!=</span> <span class="st">"NK"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(condition)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">condition</th>
<th style="text-align: right;">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Autol Balb-CD4</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Autol Balb-CD8</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Autol C57 CD4</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Autol C57-CD8</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mixed CD4 HLA B</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mixed CD4-HLA D</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mixed CD8-HLA B</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mixed CD8-HLA D</td>
<td style="text-align: right;">3</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Reject CD4</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">Reject CD8</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: left;">WT control CD4</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="even">
<td style="text-align: left;">WT control CD8</td>
<td style="text-align: right;">2</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>I started the analysis by comparing the conditions, keeping the comparisons intra strain and intra cell type. That means no comparing Black 6 cells to Balb/c or CD4 to CD8 and vice versa.</p>
<p>We wanted the differential gene expression to be due solely to the treatment conditions and not introduce possible confounders of different mouse strains and cell types.</p>
<p>Here are tables of the conditions separated by H2K:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Look at each condition and its corresponding H2K status</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>meta <span class="sc">|&gt;</span> </span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(cell_type <span class="sc">!=</span> <span class="st">"NK"</span> <span class="sc">&amp;</span> H2K <span class="sc">==</span> <span class="st">"b"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(condition, H2K, cell_type) <span class="sc">|&gt;</span> </span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">condition</th>
<th style="text-align: left;">H2K</th>
<th style="text-align: left;">cell_type</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: left;">Reject CD4</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD4</td>
</tr>
<tr class="even">
<td style="text-align: left;">3</td>
<td style="text-align: left;">Autol C57 CD4</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5</td>
<td style="text-align: left;">Mixed CD4 HLA B</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD4</td>
</tr>
<tr class="even">
<td style="text-align: left;">8</td>
<td style="text-align: left;">Reject CD8</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">10</td>
<td style="text-align: left;">Autol C57-CD8</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD8</td>
</tr>
<tr class="even">
<td style="text-align: left;">12</td>
<td style="text-align: left;">Mixed CD8-HLA B</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">15</td>
<td style="text-align: left;">WT control CD8</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD8</td>
</tr>
<tr class="even">
<td style="text-align: left;">17</td>
<td style="text-align: left;">WT control CD4</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD4</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>meta <span class="sc">|&gt;</span> </span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(cell_type <span class="sc">!=</span> <span class="st">"NK"</span> <span class="sc">&amp;</span> H2K <span class="sc">==</span> <span class="st">"d"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(condition, H2K, cell_type) <span class="sc">|&gt;</span> </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">condition</th>
<th style="text-align: left;">H2K</th>
<th style="text-align: left;">cell_type</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: left;">Autol Balb-CD4</td>
<td style="text-align: left;">d</td>
<td style="text-align: left;">CD4</td>
</tr>
<tr class="even">
<td style="text-align: left;">4</td>
<td style="text-align: left;">Mixed CD4-HLA D</td>
<td style="text-align: left;">d</td>
<td style="text-align: left;">CD4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">7</td>
<td style="text-align: left;">Autol Balb-CD8</td>
<td style="text-align: left;">d</td>
<td style="text-align: left;">CD8</td>
</tr>
<tr class="even">
<td style="text-align: left;">10</td>
<td style="text-align: left;">Mixed CD8-HLA D</td>
<td style="text-align: left;">d</td>
<td style="text-align: left;">CD8</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Since the rejection mice are H2Kb and the extra control data are also H2Kb, most of our comparisons were among the H2Kb cells. Below is a table of all the DESeq sets we wanted to perform.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Code%20Review%20Figures/Condtion%20DESeq%20table.png" class="img-fluid figure-img"></p>
</figure>
</div>
<p>14 total DESeq runs. Tzu always tells me that if I have to do something 3 or more times, it’s time for a function.</p>
<p>First, I set up the common variables needed to run DESeq.</p>
</section>
<section id="deseq-and-genetonic-object-creation" class="level1">
<h1>DESeq and GeneTonic object creation</h1>
<section id="organization-creating-result-folders" class="level2">
<h2 class="anchored" data-anchor-id="organization-creating-result-folders">Organization: Creating Result Folders</h2>
<p>I created two results for each comparison: DESeq results tables (.csv) GeneTonic objects (.Rdata)</p>
<p>The original script that did this was created 2024_01_10. Kept folder names consistent for this walkthrough.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>folder.names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"2024_01_10_results"</span>, <span class="st">"2024_01_10_results/deseq_results"</span>, <span class="st">"2024_01_10_results/genetonic_results"</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> folder.names) {</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">exists</span>(i)){</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dir.create</span>(i)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(i): '2024_01_10_results' already exists</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(i): '2024_01_10_results\deseq_results' already exists</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(i): '2024_01_10_results\genetonic_results' already exists</code></pre>
</div>
</div>
</section>
<section id="creating-counts-matrix" class="level2">
<h2 class="anchored" data-anchor-id="creating-counts-matrix">Creating counts matrix</h2>
<section id="create-and-saveload-gtf_df" class="level3">
<h3 class="anchored" data-anchor-id="create-and-saveload-gtf_df">Create and save/load gtf_df</h3>
<p>Import the mouse gene feature file</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>gtf <span class="ot">&lt;-</span>  rtracklayer<span class="sc">::</span><span class="fu">import</span>(<span class="st">"data/Mus_musculus.GRCm39.110.gtf"</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>gtf_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(gtf)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gtf_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 5%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 5%">
<col style="width: 4%">
<col style="width: 9%">
<col style="width: 3%">
<col style="width: 5%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 4%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">seqnames</th>
<th style="text-align: right;">start</th>
<th style="text-align: right;">end</th>
<th style="text-align: right;">width</th>
<th style="text-align: left;">strand</th>
<th style="text-align: left;">source</th>
<th style="text-align: left;">type</th>
<th style="text-align: right;">score</th>
<th style="text-align: right;">phase</th>
<th style="text-align: left;">gene_id</th>
<th style="text-align: left;">gene_version</th>
<th style="text-align: left;">gene_name</th>
<th style="text-align: left;">gene_source</th>
<th style="text-align: left;">gene_biotype</th>
<th style="text-align: left;">transcript_id</th>
<th style="text-align: left;">transcript_version</th>
<th style="text-align: left;">transcript_name</th>
<th style="text-align: left;">transcript_source</th>
<th style="text-align: left;">transcript_biotype</th>
<th style="text-align: left;">tag</th>
<th style="text-align: left;">transcript_support_level</th>
<th style="text-align: left;">exon_number</th>
<th style="text-align: left;">exon_id</th>
<th style="text-align: left;">exon_version</th>
<th style="text-align: left;">ccds_id</th>
<th style="text-align: left;">protein_id</th>
<th style="text-align: left;">protein_version</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: right;">108344807</td>
<td style="text-align: right;">108347562</td>
<td style="text-align: right;">2756</td>
<td style="text-align: left;">+</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">gene</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">ENSMUSG00000104478</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">Gm38212</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">TEC</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">1</td>
<td style="text-align: right;">108344807</td>
<td style="text-align: right;">108347562</td>
<td style="text-align: right;">2756</td>
<td style="text-align: left;">+</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">transcript</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">ENSMUSG00000104478</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">Gm38212</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">TEC</td>
<td style="text-align: left;">ENSMUST00000194081</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">Gm38212-201</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">TEC</td>
<td style="text-align: left;">Ensembl_canonical</td>
<td style="text-align: left;">NA (assigned to previous version 1)</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: right;">108344807</td>
<td style="text-align: right;">108347562</td>
<td style="text-align: right;">2756</td>
<td style="text-align: left;">+</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">exon</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">ENSMUSG00000104478</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">Gm38212</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">TEC</td>
<td style="text-align: left;">ENSMUST00000194081</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">Gm38212-201</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">TEC</td>
<td style="text-align: left;">Ensembl_canonical</td>
<td style="text-align: left;">NA (assigned to previous version 1)</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">ENSMUSE00001337335</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">1</td>
<td style="text-align: right;">6980784</td>
<td style="text-align: right;">6981446</td>
<td style="text-align: right;">663</td>
<td style="text-align: left;">+</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">gene</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">ENSMUSG00000104385</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">Gm7449</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">processed_pseudogene</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: right;">6980784</td>
<td style="text-align: right;">6981446</td>
<td style="text-align: right;">663</td>
<td style="text-align: left;">+</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">transcript</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">ENSMUSG00000104385</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">Gm7449</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">processed_pseudogene</td>
<td style="text-align: left;">ENSMUST00000194393</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">Gm7449-201</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">processed_pseudogene</td>
<td style="text-align: left;">Ensembl_canonical</td>
<td style="text-align: left;">NA (assigned to previous version 1)</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">1</td>
<td style="text-align: right;">6980784</td>
<td style="text-align: right;">6981446</td>
<td style="text-align: right;">663</td>
<td style="text-align: left;">+</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">exon</td>
<td style="text-align: right;">NA</td>
<td style="text-align: right;">NA</td>
<td style="text-align: left;">ENSMUSG00000104385</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">Gm7449</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">processed_pseudogene</td>
<td style="text-align: left;">ENSMUST00000194393</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">Gm7449-201</td>
<td style="text-align: left;">havana</td>
<td style="text-align: left;">processed_pseudogene</td>
<td style="text-align: left;">Ensembl_canonical</td>
<td style="text-align: left;">NA (assigned to previous version 1)</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">ENSMUSE00001339893</td>
<td style="text-align: left;">2</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Create a look-up-table matching gene id to all its transcript ids:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>tx2gene.df <span class="ot">&lt;-</span> gtf_df <span class="sc">|&gt;</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(transcript_id, gene_id) <span class="sc">|&gt;</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(transcript_id, gene_id) <span class="sc">|&gt;</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>() <span class="ot">-&gt;</span> tx2gene.df</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tx2gene.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">transcript_id</th>
<th style="text-align: left;">gene_id</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ENSMUST00000194081</td>
<td style="text-align: left;">ENSMUSG00000104478</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUST00000194393</td>
<td style="text-align: left;">ENSMUSG00000104385</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUST00000185509</td>
<td style="text-align: left;">ENSMUSG00000101231</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUST00000194605</td>
<td style="text-align: left;">ENSMUSG00000102135</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUST00000191703</td>
<td style="text-align: left;">ENSMUSG00000103282</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUST00000191467</td>
<td style="text-align: left;">ENSMUSG00000101097</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
<section id="read-in-salmon-quant.sf-files-using-a-loop" class="level3">
<h3 class="anchored" data-anchor-id="read-in-salmon-quant.sf-files-using-a-loop">Read in salmon quant.sf files using a loop</h3>
<p>This chunk reads the first <code>quant.sf</code> and joins it with the <code>tx2gene.df</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span>  <span class="fu">read.delim</span>(<span class="fu">paste0</span>(<span class="st">"data/raw_data/salmon_output_files/"</span>,</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>                          meta<span class="sc">$</span>folder_name[<span class="dv">1</span>], <span class="st">"/quant.sf"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Name =</span> <span class="fu">str_sub</span>(Name, <span class="at">end=</span><span class="sc">-</span><span class="dv">3</span>)) <span class="sc">|&gt;</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(tx2gene.df, <span class="at">by=</span><span class="fu">c</span>(<span class="st">"Name"</span><span class="ot">=</span><span class="st">"transcript_id"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_id) <span class="sc">|&gt;</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">NumReads=</span><span class="fu">as.integer</span>(<span class="fu">sum</span>(NumReads)))</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tmp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">gene_id</th>
<th style="text-align: right;">NumReads</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000000001</td>
<td style="text-align: right;">4654</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000000003</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000000028</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000000031</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000000037</td>
<td style="text-align: right;">6</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000000049</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Now we pre-define an empty count matrix to the correct dimensions. Then we will perform a loop to read the quant files and fill in the matrix one column at a time.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>sf.df <span class="ot">&lt;-</span>  <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="dv">0</span>, </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">nrow =</span> <span class="fu">length</span>(tmp<span class="sc">$</span>gene_id),</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">ncol =</span> <span class="fu">length</span>(meta<span class="sc">$</span>folder_name)))</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(sf.df) <span class="ot">&lt;-</span>  meta<span class="sc">$</span>folder_name</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sf.df) <span class="ot">&lt;-</span>  tmp<span class="sc">$</span>gene_id</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(meta<span class="sc">$</span>folder_name)){</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"Now reading..."</span>, meta<span class="sc">$</span>folder_name[i], <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  sf.df[,i] <span class="ot">&lt;-</span>  <span class="fu">read.delim</span>(<span class="fu">paste0</span>(<span class="st">"data/raw_data/salmon_output_files/"</span>, meta<span class="sc">$</span>folder_name[i], <span class="st">"/quant.sf"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Name =</span> <span class="fu">str_sub</span>(Name, <span class="at">end =</span> <span class="sc">-</span><span class="dv">3</span>)) <span class="sc">|&gt;</span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">inner_join</span>(tx2gene.df, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Name"</span> <span class="ot">=</span> <span class="st">"transcript_id"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(gene_id) <span class="sc">|&gt;</span></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">NumReads =</span> <span class="fu">as.integer</span>(<span class="fu">sum</span>(NumReads))) <span class="sc">|&gt;</span></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(NumReads)</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Now reading... 52-CD4_S37_L002 
Now reading... 53-CD4_S39_L002 
Now reading... 55-CD4_S41_L002 
Now reading... 56-CD4_S43_L002 
Now reading... 61-CD4_S57_L002 
Now reading... 62-CD4_S59_L002 
Now reading... 63-CD4_S61_L002 
Now reading... 57-CD4-B_S45_L002 
Now reading... 58-CD4-B_S49_L002 
Now reading... 60-CD4-B_S53_L002 
Now reading... 57-CD4-D_S46_L002 
Now reading... 58-CD4-D_S50_L002 
Now reading... 60-CD4-D_S54_L002 
Now reading... 52-CD8_S38_L002 
Now reading... 53-CD8_S40_L002 
Now reading... 55-CD8_S42_L002 
Now reading... 56-CD8_S44_L002 
Now reading... 61-CD8_S58_L002 
Now reading... 62-CD8_S60_L002 
Now reading... 63-CD8_S62_L002 
Now reading... 57-CD8-B_S47_L002 
Now reading... 58-CD8-B_S51_L002 
Now reading... 60-CD8-B_S55_L002 
Now reading... 57-CD8-D_S48_L002 
Now reading... 58-CD8-D_S52_L002 
Now reading... 60-CD8-D_S56_L002 
Now reading... NK-1_S32_L002 
Now reading... NK-2B_S33_L002 
Now reading... NK-2D_S34_L002 
Now reading... NK-3_S35_L002 
Now reading... NK-4_S36_L002 
Now reading... SRR6927755 
Now reading... SRR6927757 
Now reading... SRR6927754 
Now reading... SRR6927756 </code></pre>
</div>
</div>
<p>Take a look at the sf.df</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sf.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 2%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">52-CD4_S37_L002</th>
<th style="text-align: right;">53-CD4_S39_L002</th>
<th style="text-align: right;">55-CD4_S41_L002</th>
<th style="text-align: right;">56-CD4_S43_L002</th>
<th style="text-align: right;">61-CD4_S57_L002</th>
<th style="text-align: right;">62-CD4_S59_L002</th>
<th style="text-align: right;">63-CD4_S61_L002</th>
<th style="text-align: right;">57-CD4-B_S45_L002</th>
<th style="text-align: right;">58-CD4-B_S49_L002</th>
<th style="text-align: right;">60-CD4-B_S53_L002</th>
<th style="text-align: right;">57-CD4-D_S46_L002</th>
<th style="text-align: right;">58-CD4-D_S50_L002</th>
<th style="text-align: right;">60-CD4-D_S54_L002</th>
<th style="text-align: right;">52-CD8_S38_L002</th>
<th style="text-align: right;">53-CD8_S40_L002</th>
<th style="text-align: right;">55-CD8_S42_L002</th>
<th style="text-align: right;">56-CD8_S44_L002</th>
<th style="text-align: right;">61-CD8_S58_L002</th>
<th style="text-align: right;">62-CD8_S60_L002</th>
<th style="text-align: right;">63-CD8_S62_L002</th>
<th style="text-align: right;">57-CD8-B_S47_L002</th>
<th style="text-align: right;">58-CD8-B_S51_L002</th>
<th style="text-align: right;">60-CD8-B_S55_L002</th>
<th style="text-align: right;">57-CD8-D_S48_L002</th>
<th style="text-align: right;">58-CD8-D_S52_L002</th>
<th style="text-align: right;">60-CD8-D_S56_L002</th>
<th style="text-align: right;">NK-1_S32_L002</th>
<th style="text-align: right;">NK-2B_S33_L002</th>
<th style="text-align: right;">NK-2D_S34_L002</th>
<th style="text-align: right;">NK-3_S35_L002</th>
<th style="text-align: right;">NK-4_S36_L002</th>
<th style="text-align: right;">SRR6927755</th>
<th style="text-align: right;">SRR6927757</th>
<th style="text-align: right;">SRR6927754</th>
<th style="text-align: right;">SRR6927756</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000000001</td>
<td style="text-align: right;">4654</td>
<td style="text-align: right;">4963</td>
<td style="text-align: right;">4932</td>
<td style="text-align: right;">5720</td>
<td style="text-align: right;">4405</td>
<td style="text-align: right;">4642</td>
<td style="text-align: right;">4828</td>
<td style="text-align: right;">4498</td>
<td style="text-align: right;">4109</td>
<td style="text-align: right;">4277</td>
<td style="text-align: right;">4585</td>
<td style="text-align: right;">5380</td>
<td style="text-align: right;">3781</td>
<td style="text-align: right;">4023</td>
<td style="text-align: right;">4508</td>
<td style="text-align: right;">5003</td>
<td style="text-align: right;">4604</td>
<td style="text-align: right;">4027</td>
<td style="text-align: right;">3956</td>
<td style="text-align: right;">4681</td>
<td style="text-align: right;">3654</td>
<td style="text-align: right;">2435</td>
<td style="text-align: right;">3165</td>
<td style="text-align: right;">2897</td>
<td style="text-align: right;">3492</td>
<td style="text-align: right;">3470</td>
<td style="text-align: right;">5601</td>
<td style="text-align: right;">5302</td>
<td style="text-align: right;">5956</td>
<td style="text-align: right;">6435</td>
<td style="text-align: right;">6186</td>
<td style="text-align: right;">1314</td>
<td style="text-align: right;">1840</td>
<td style="text-align: right;">1360</td>
<td style="text-align: right;">2371</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000000003</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000000028</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">17</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000000031</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000000037</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">25</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">27</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">30</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">14</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">26</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000000049</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(gtf, tx2gene.df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="deseq-analysis---single-comparison-walk-through" class="level2">
<h2 class="anchored" data-anchor-id="deseq-analysis---single-comparison-walk-through">DESeq analysis - single comparison walk through</h2>
<p>In a smaller bulk RNAseq data, the counts table would be filtered for low reads as the next step.</p>
<p>However, this data has multiple cell types (CD4, CD8, NK) and other variables like mouse strain (H2Kb/H2Kd).</p>
<p>I needed to create a subset of the data that contains the appropriate samples for my comparison of interest before filtering for reads.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>meta <span class="sc">|&gt;</span> </span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(cell_type <span class="sc">!=</span> <span class="st">"NK"</span> <span class="sc">&amp;</span> H2K <span class="sc">==</span> <span class="st">"b"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(condition, H2K, cell_type) <span class="sc">|&gt;</span> </span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">condition</th>
<th style="text-align: left;">H2K</th>
<th style="text-align: left;">cell_type</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">1</td>
<td style="text-align: left;">Reject CD4</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD4</td>
</tr>
<tr class="even">
<td style="text-align: left;">3</td>
<td style="text-align: left;">Autol C57 CD4</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">5</td>
<td style="text-align: left;">Mixed CD4 HLA B</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD4</td>
</tr>
<tr class="even">
<td style="text-align: left;">8</td>
<td style="text-align: left;">Reject CD8</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">10</td>
<td style="text-align: left;">Autol C57-CD8</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD8</td>
</tr>
<tr class="even">
<td style="text-align: left;">12</td>
<td style="text-align: left;">Mixed CD8-HLA B</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD8</td>
</tr>
<tr class="odd">
<td style="text-align: left;">15</td>
<td style="text-align: left;">WT control CD8</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD8</td>
</tr>
<tr class="even">
<td style="text-align: left;">17</td>
<td style="text-align: left;">WT control CD4</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">CD4</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>This comparison will use the following:</p>
<ul>
<li><p>condition 1 : <code>Mixed CD4 HLA B</code></p></li>
<li><p>condition 2 : <code>Reject CD4</code></p></li>
</ul>
<p>I used a shorthand name when writing results files:</p>
<ul>
<li><p>name 1 : <code>mix.4.b</code></p></li>
<li><p>name 2 : <code>rej.4.b</code></p></li>
</ul>
<p>The code chunk below creates a vector containing the names of the sample in our comparison. It uses the vector to subset the counts matrix and then filters out any genes that have less than 10 reads in any of the subset samples.</p>
<p>Lastly, we subset the metadata for use in our DESeq object later.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset metadata to the conditions of interest and pull out those sample names</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>subset.vector <span class="ot">&lt;-</span> meta <span class="sc">|&gt;</span> </span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(condition <span class="sc">==</span> <span class="st">"Mixed CD4 HLA B"</span> <span class="sc">|</span> condition <span class="sc">==</span> <span class="st">"Reject CD4"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(folder_name)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>subset.vector</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "52-CD4_S37_L002"   "53-CD4_S39_L002"   "57-CD4-B_S45_L002"
[4] "58-CD4-B_S49_L002" "60-CD4-B_S53_L002"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset the sf.df</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>subset.sf.df <span class="ot">&lt;-</span> sf.df <span class="sc">|&gt;</span> </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(subset.vector)) </span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Use subset.vector to filter the original sf.df object</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>subset.sf.df <span class="ot">&lt;-</span> subset.sf.df <span class="sc">|&gt;</span> </span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">row.sum =</span> <span class="fu">apply</span>(subset.sf.df, <span class="dv">1</span>, <span class="cf">function</span>(x) <span class="fu">sum</span>(x <span class="sc">&gt;</span> <span class="dv">10</span>))) <span class="sc">|&gt;</span> </span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># All values must be more than 10 (set to x if only x value is satisfied)</span></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(row.sum <span class="sc">==</span> <span class="fu">length</span>(subset.vector)) <span class="sc">|&gt;</span> </span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove row.sum, it is not needing moving forward</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>row.sum)</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Filtering metadata so only relevant samples are put into the DESeq object</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>subset_meta <span class="ot">&lt;-</span> meta <span class="sc">|&gt;</span> </span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(folder_name <span class="sc">%in%</span> subset.vector)</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a>subset_meta</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 2%">
<col style="width: 7%">
<col style="width: 10%">
<col style="width: 12%">
<col style="width: 7%">
<col style="width: 13%">
<col style="width: 7%">
<col style="width: 10%">
<col style="width: 3%">
<col style="width: 9%">
<col style="width: 7%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: right;">X</th>
<th style="text-align: left;">sample_id</th>
<th style="text-align: left;">revised_id</th>
<th style="text-align: left;">condition</th>
<th style="text-align: left;">group</th>
<th style="text-align: left;">folder_name</th>
<th style="text-align: left;">cell_type</th>
<th style="text-align: left;">mouse_number</th>
<th style="text-align: left;">H2K</th>
<th style="text-align: left;">recip_mouse</th>
<th style="text-align: right;">radiation</th>
<th style="text-align: right;">replicate</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">1</td>
<td style="text-align: left;">CD4_52</td>
<td style="text-align: left;">CD4_reject_1</td>
<td style="text-align: left;">Reject CD4</td>
<td style="text-align: left;">Rejection</td>
<td style="text-align: left;">52-CD4_S37_L002</td>
<td style="text-align: left;">CD4</td>
<td style="text-align: left;">52</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">b6</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: right;">2</td>
<td style="text-align: left;">CD4_53</td>
<td style="text-align: left;">CD4_reject_2</td>
<td style="text-align: left;">Reject CD4</td>
<td style="text-align: left;">Rejection</td>
<td style="text-align: left;">53-CD4_S39_L002</td>
<td style="text-align: left;">CD4</td>
<td style="text-align: left;">53</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">b6</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: right;">8</td>
<td style="text-align: left;">CD4_B_57</td>
<td style="text-align: left;">CD4_mix_B_1</td>
<td style="text-align: left;">Mixed CD4 HLA B</td>
<td style="text-align: left;">Mixed</td>
<td style="text-align: left;">57-CD4-B_S45_L002</td>
<td style="text-align: left;">CD4</td>
<td style="text-align: left;">57</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">b6</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">1</td>
</tr>
<tr class="even">
<td style="text-align: right;">9</td>
<td style="text-align: left;">CD4_B_58</td>
<td style="text-align: left;">CD4_mix_B_2</td>
<td style="text-align: left;">Mixed CD4 HLA B</td>
<td style="text-align: left;">Mixed</td>
<td style="text-align: left;">58-CD4-B_S49_L002</td>
<td style="text-align: left;">CD4</td>
<td style="text-align: left;">58</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">b6</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">2</td>
</tr>
<tr class="odd">
<td style="text-align: right;">10</td>
<td style="text-align: left;">CD4_B_60</td>
<td style="text-align: left;">CD4_mix_B_3</td>
<td style="text-align: left;">Mixed CD4 HLA B</td>
<td style="text-align: left;">Mixed</td>
<td style="text-align: left;">60-CD4-B_S53_L002</td>
<td style="text-align: left;">CD4</td>
<td style="text-align: left;">60</td>
<td style="text-align: left;">b</td>
<td style="text-align: left;">b6</td>
<td style="text-align: right;">6</td>
<td style="text-align: right;">3</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<section id="salmon-matrix-qc-check-and-deseqdataset-dds-creation" class="level3">
<h3 class="anchored" data-anchor-id="salmon-matrix-qc-check-and-deseqdataset-dds-creation">Salmon matrix, QC check, and DESeqDataSet (dds) creation</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>subset.salmon.matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(subset.sf.df)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>subset.coldata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample =</span> subset_meta<span class="sc">$</span>folder_name,</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">condition =</span> <span class="fu">as.factor</span>(subset_meta<span class="sc">$</span>condition),</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">row.names =</span> <span class="st">"sample"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Quick row and column check</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(subset.coldata) <span class="sc">%in%</span> <span class="fu">colnames</span>(subset.salmon.matrix))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>subset.se <span class="ot">&lt;-</span> <span class="fu">SummarizedExperiment</span>(<span class="at">assays =</span> <span class="fu">list</span>(<span class="at">counts =</span> subset.salmon.matrix),</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>                           <span class="at">colData =</span> subset.coldata)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>subset.se</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: SummarizedExperiment 
dim: 9958 5 
metadata(0):
assays(1): counts
rownames(9958): ENSMUSG00000000001 ENSMUSG00000000056 ...
  ENSMUSG00002076161 ENSMUSG00002076556
rowData names(0):
colnames(5): 52-CD4_S37_L002 53-CD4_S39_L002 57-CD4-B_S45_L002
  58-CD4-B_S49_L002 60-CD4-B_S53_L002
colData names(1): condition</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>subset.dds <span class="ot">=</span> <span class="fu">DESeqDataSet</span>(subset.se, <span class="at">design =</span> <span class="sc">~</span> condition)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
</div>
</section>
<section id="run-deseq" class="level3">
<h3 class="anchored" data-anchor-id="run-deseq">Run DESeq</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>subset.dds <span class="ot">=</span> <span class="fu">DESeq</span>(subset.dds)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>  Note: levels of factors in the design contain characters other than
  letters, numbers, '_' and '.'. It is recommended (but not required) to use
  only letters, numbers, and delimiters '_' or '.', as these are safe characters
  for column names in R. [This is a message, not a warning or an error]</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a results table of our comparison of interest</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>subset.res <span class="ot">&lt;-</span> <span class="fu">results</span>(subset.dds,</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"condition"</span>, <span class="st">"Mixed CD4 HLA B"</span>, <span class="st">"Reject CD4"</span>),</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">alpha =</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="deseq-results-in-an-annotated-table" class="level3">
<h3 class="anchored" data-anchor-id="deseq-results-in-an-annotated-table">DESeq results in an annotated table</h3>
<p>Now we turn the results object into a dataframe with gene annotation, add some extra annotation (<code>gene_symbol</code>, <code>gene_entrez</code>, <code>gene_desc</code>) and use <code>left_join</code> to include the raw count data. Converting the Ensembl ID row names of <code>subset.sf.df</code> allows us to join the raw counts to the result dataframe by the shared Ensembl ID column called <code>gene_id</code> in both dataframes.</p>
<p>It is good practice to include the raw counts in the results to confirm the comparisons were done correctly.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>subset.ids <span class="ot">&lt;-</span> subset.sf.df <span class="sc">|&gt;</span> </span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"gene_id"</span>)</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>subset.anno <span class="ot">&lt;-</span> <span class="fu">deseqresult2df</span>(subset.res) <span class="sc">|&gt;</span> </span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene_id =</span> id) <span class="sc">|&gt;</span> </span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_symbol =</span> <span class="fu">mapIds</span>(org.Mm.eg.db, </span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">keys =</span> gene_id, </span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>                              <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>, </span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>                              <span class="at">column =</span> <span class="st">"SYMBOL"</span>, </span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>                              <span class="at">multiVals =</span> <span class="st">"first"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_entrez =</span> <span class="fu">mapIds</span>(org.Mm.eg.db, </span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>                              <span class="at">keys =</span> gene_id, </span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>                              <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>, </span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>                              <span class="at">column =</span> <span class="st">"ENTREZID"</span>, </span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>                              <span class="at">multiVals =</span> <span class="st">"first"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_desc =</span> <span class="fu">mapIds</span>(org.Mm.eg.db, </span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>                            <span class="at">keys =</span> gene_id, </span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a>                            <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>, </span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>                            <span class="at">column =</span> <span class="st">"GENENAME"</span>, </span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a>                            <span class="at">multiVals =</span> <span class="st">"first"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(gene_symbol, gene_entrez, gene_desc, <span class="at">.after =</span> gene_id) <span class="sc">|&gt;</span> </span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(subset.ids, </span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene_id"</span><span class="ot">=</span><span class="st">"gene_id"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns
'select()' returned 1:many mapping between keys and columns
'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(subset.anno)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 8%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 15%">
<col style="width: 4%">
<col style="width: 6%">
<col style="width: 4%">
<col style="width: 4%">
<col style="width: 3%">
<col style="width: 2%">
<col style="width: 7%">
<col style="width: 7%">
<col style="width: 8%">
<col style="width: 8%">
<col style="width: 8%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">gene_id</th>
<th style="text-align: left;">gene_symbol</th>
<th style="text-align: left;">gene_entrez</th>
<th style="text-align: left;">gene_desc</th>
<th style="text-align: right;">baseMean</th>
<th style="text-align: right;">log2FoldChange</th>
<th style="text-align: right;">lfcSE</th>
<th style="text-align: right;">stat</th>
<th style="text-align: right;">pvalue</th>
<th style="text-align: right;">padj</th>
<th style="text-align: right;">52-CD4_S37_L002</th>
<th style="text-align: right;">53-CD4_S39_L002</th>
<th style="text-align: right;">57-CD4-B_S45_L002</th>
<th style="text-align: right;">58-CD4-B_S49_L002</th>
<th style="text-align: right;">60-CD4-B_S53_L002</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000052684</td>
<td style="text-align: left;">Jun</td>
<td style="text-align: left;">16476</td>
<td style="text-align: left;">jun proto-oncogene</td>
<td style="text-align: right;">9729.037</td>
<td style="text-align: right;">3.581938</td>
<td style="text-align: right;">0.1042002</td>
<td style="text-align: right;">34.37556</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1375</td>
<td style="text-align: right;">1424</td>
<td style="text-align: right;">17108</td>
<td style="text-align: right;">13530</td>
<td style="text-align: right;">13801</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000003545</td>
<td style="text-align: left;">Fosb</td>
<td style="text-align: left;">14282</td>
<td style="text-align: left;">FBJ osteosarcoma oncogene B</td>
<td style="text-align: right;">9917.000</td>
<td style="text-align: right;">6.911367</td>
<td style="text-align: right;">0.2895151</td>
<td style="text-align: right;">23.87222</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">199</td>
<td style="text-align: right;">96</td>
<td style="text-align: right;">17749</td>
<td style="text-align: right;">16200</td>
<td style="text-align: right;">13398</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000003032</td>
<td style="text-align: left;">Klf4</td>
<td style="text-align: left;">16600</td>
<td style="text-align: left;">Kruppel-like factor 4 (gut)</td>
<td style="text-align: right;">1248.193</td>
<td style="text-align: right;">5.787720</td>
<td style="text-align: right;">0.3489098</td>
<td style="text-align: right;">16.58801</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">44</td>
<td style="text-align: right;">37</td>
<td style="text-align: right;">1462</td>
<td style="text-align: right;">2268</td>
<td style="text-align: right;">2079</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000021250</td>
<td style="text-align: left;">Fos</td>
<td style="text-align: left;">14281</td>
<td style="text-align: left;">FBJ osteosarcoma oncogene</td>
<td style="text-align: right;">11695.332</td>
<td style="text-align: right;">4.140927</td>
<td style="text-align: right;">0.2605069</td>
<td style="text-align: right;">15.89565</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1095</td>
<td style="text-align: right;">1231</td>
<td style="text-align: right;">13292</td>
<td style="text-align: right;">21027</td>
<td style="text-align: right;">18729</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000027907</td>
<td style="text-align: left;">S100a11</td>
<td style="text-align: left;">20195</td>
<td style="text-align: left;">S100 calcium binding protein A11</td>
<td style="text-align: right;">5223.092</td>
<td style="text-align: right;">1.648460</td>
<td style="text-align: right;">0.1085009</td>
<td style="text-align: right;">15.19306</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2499</td>
<td style="text-align: right;">2495</td>
<td style="text-align: right;">7961</td>
<td style="text-align: right;">6072</td>
<td style="text-align: right;">6752</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000001020</td>
<td style="text-align: left;">S100a4</td>
<td style="text-align: left;">20198</td>
<td style="text-align: left;">S100 calcium binding protein A4</td>
<td style="text-align: right;">4297.850</td>
<td style="text-align: right;">2.837430</td>
<td style="text-align: right;">0.1970625</td>
<td style="text-align: right;">14.39863</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1037</td>
<td style="text-align: right;">960</td>
<td style="text-align: right;">8564</td>
<td style="text-align: right;">5277</td>
<td style="text-align: right;">5309</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<p>Save the DESeq results separately.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(subset.anno, <span class="at">file =</span> <span class="st">"2024_01_10_results/DESeq_results/mix.4.b.VS.rej.4.b.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="genetonic" class="level2">
<h2 class="anchored" data-anchor-id="genetonic">GeneTonic</h2>
<section id="genetonic-prep" class="level3">
<h3 class="anchored" data-anchor-id="genetonic-prep">GeneTonic Prep</h3>
<p>GeneTonic needs:</p>
<ul>
<li><p>A DESeqDataSet (<code>subset.dds</code>) object, which we already created.</p></li>
<li><p>A DESEq results object (<code>res</code>), which we also already created.</p></li>
<li><p>An enriched results object (a DESeq results object filtered to significant genes, <code>.topgo</code>)</p></li>
<li><p>Background gene ids (<code>bg_ids</code>)</p></li>
<li><p>A gene annotation object (<code>anno</code>)</p></li>
</ul>
<section id="enriched-results-object" class="level4">
<h4 class="anchored" data-anchor-id="enriched-results-object">Enriched results object</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>enrich.of.interest <span class="ot">&lt;-</span> <span class="fu">deseqresult2df</span>(subset.res) <span class="sc">|&gt;</span> </span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Symbol =</span> <span class="fu">mapIds</span>(org.Mm.eg.db, </span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">keys =</span> id, </span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>, </span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">column =</span> <span class="st">"SYMBOL"</span>, </span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">multiVals =</span> <span class="st">"first"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span> </span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(Symbol)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(enrich.of.interest)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 921</code></pre>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(enrich.of.interest)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>ENSMUSG00000052684 ENSMUSG00000003545 ENSMUSG00000003032 ENSMUSG00000021250 
             "Jun"             "Fosb"             "Klf4"              "Fos" 
ENSMUSG00000027907 ENSMUSG00000001020 
         "S100a11"           "S100a4" </code></pre>
</div>
</div>
</section>
<section id="background-gene-symbols" class="level4">
<h4 class="anchored" data-anchor-id="background-gene-symbols">Background gene symbols</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>bg_ids <span class="ot">&lt;-</span> <span class="fu">deseqresult2df</span>(subset.res) <span class="sc">|&gt;</span> </span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Symbol =</span> <span class="fu">mapIds</span>(org.Mm.eg.db, </span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a>                         <span class="at">keys =</span> id, </span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a>                         <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>, </span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a>                         <span class="at">column =</span> <span class="st">"SYMBOL"</span>, </span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a>                         <span class="at">multiVals =</span> <span class="st">"first"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(Symbol)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(bg_ids)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 9958</code></pre>
</div>
</div>
</section>
<section id="gene-annotation-object" class="level4">
<h4 class="anchored" data-anchor-id="gene-annotation-object">Gene annotation object</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>sf.df.anno <span class="ot">&lt;-</span> <span class="fu">deseqresult2df</span>(subset.res) <span class="sc">|&gt;</span> </span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene_id =</span> id) <span class="sc">|&gt;</span> </span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_name =</span> <span class="fu">mapIds</span>(org.Mm.eg.db, </span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>                            <span class="at">keys =</span> gene_id, </span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a>                            <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>, </span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>                            <span class="at">column =</span> <span class="st">"SYMBOL"</span>, </span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>                            <span class="at">multiVals =</span> <span class="st">"first"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id, gene_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sf.df.anno)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">gene_id</th>
<th style="text-align: left;">gene_name</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000052684</td>
<td style="text-align: left;">ENSMUSG00000052684</td>
<td style="text-align: left;">Jun</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000003545</td>
<td style="text-align: left;">ENSMUSG00000003545</td>
<td style="text-align: left;">Fosb</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000003032</td>
<td style="text-align: left;">ENSMUSG00000003032</td>
<td style="text-align: left;">Klf4</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000021250</td>
<td style="text-align: left;">ENSMUSG00000021250</td>
<td style="text-align: left;">Fos</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000027907</td>
<td style="text-align: left;">ENSMUSG00000027907</td>
<td style="text-align: left;">S100a11</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000001020</td>
<td style="text-align: left;">ENSMUSG00000001020</td>
<td style="text-align: left;">S100a4</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
</section>
<section id="perform-pathway-analysis" class="level3">
<h3 class="anchored" data-anchor-id="perform-pathway-analysis">Perform Pathway Analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>subset.topgo <span class="ot">=</span> pcaExplorer<span class="sc">::</span><span class="fu">topGOtable</span>(enrich.of.interest,</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>                                       bg_ids,</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">ontology =</span> <span class="st">"BP"</span>,</span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">mapping =</span> <span class="st">"org.Mm.eg.db"</span>,</span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">geneID =</span> <span class="st">"symbol"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Building most specific GOs .....</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 9852 GO terms found. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Build GO DAG topology ..........</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 13357 GO terms and 30096 relations. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Annotating nodes ...............</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>    ( 8902 genes annotated to the GO terms. )</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
             -- Elim Algorithm -- 

         the algorithm is scoring 4573 nontrivial nodes
         parameters: 
             test statistic: fisher
             cutOff: 0.01</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 17:  1 nodes to be scored    (0 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 16:  12 nodes to be scored   (0 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 15:  29 nodes to be scored   (10 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 14:  52 nodes to be scored   (48 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 13:  85 nodes to be scored   (247 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 12:  158 nodes to be scored  (1261 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 11:  310 nodes to be scored  (1531 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 10:  477 nodes to be scored  (1758 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 9:   646 nodes to be scored  (2041 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 8:   696 nodes to be scored  (2601 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 7:   729 nodes to be scored  (3233 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 6:   628 nodes to be scored  (3974 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 5:   408 nodes to be scored  (4488 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 4:   231 nodes to be scored  (4945 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 3:   92 nodes to be scored   (5036 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 2:   18 nodes to be scored   (5214 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
     Level 1:   1 nodes to be scored    (5214 eliminated genes)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
             -- Classic Algorithm -- 

         the algorithm is scoring 4573 nontrivial nodes
         parameters: 
             test statistic: fisher</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in pcaExplorer::topGOtable(enrich.of.interest, bg_ids, ontology = "BP",
: NAs introduced by coercion</code></pre>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Reshape for Genetonic</span></span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>subset.topgo <span class="ot">&lt;-</span> <span class="fu">shake_topGOtableResult</span>(subset.topgo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found 200 gene sets in `topGOtableResult` object.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Converting for usage in GeneTonic...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="co"># rownames(mix.4.b.vs.rej.4.topgo) = NULL</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>subset.topgo <span class="ot">&lt;-</span> <span class="fu">get_aggrscores</span>(subset.topgo, subset.res, sf.df.anno)</span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(subset.topgo)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<colgroup>
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 6%">
<col style="width: 1%">
<col style="width: 81%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
<col style="width: 1%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">gs_id</th>
<th style="text-align: left;">gs_description</th>
<th style="text-align: right;">gs_pvalue</th>
<th style="text-align: left;">gs_genes</th>
<th style="text-align: right;">gs_de_count</th>
<th style="text-align: right;">gs_bg_count</th>
<th style="text-align: right;">Expected</th>
<th style="text-align: right;">DE_count</th>
<th style="text-align: right;">z_score</th>
<th style="text-align: right;">aggr_score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">GO:0045063</td>
<td style="text-align: left;">GO:0045063</td>
<td style="text-align: left;">T-helper 1 cell differentiation</td>
<td style="text-align: right;">1.0e-07</td>
<td style="text-align: left;">Ccr2,Ccr7,Gadd45g,Hmgb1,Il18r1,Il4ra,Lef1,Nfkbiz,Sema4a,Spn,Tbx21</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">1.73</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">0.904534</td>
<td style="text-align: right;">0.5237026</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0030101</td>
<td style="text-align: left;">GO:0030101</td>
<td style="text-align: left;">natural killer cell activation</td>
<td style="text-align: right;">2.2e-06</td>
<td style="text-align: left;">Dcaf15,Gata3,Id2,Ikzf1,Il12rb1,Il18r1,Il21,Il2rb,Klrb1c,Klrk1,Lilrb4a,Lyst,Nfil3,Nkg7,Ptpn22,Rab27a,Slamf1,Slamf6,Stx11</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">61</td>
<td style="text-align: right;">5.88</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">3.441236</td>
<td style="text-align: right;">1.0714492</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0050729</td>
<td style="text-align: left;">GO:0050729</td>
<td style="text-align: left;">positive regulation of inflammatory response</td>
<td style="text-align: right;">2.6e-06</td>
<td style="text-align: left;">Abcc1,Adam8,Aim2,Camk2n1,Casp1,Ccl5,Ccn4,Ccr2,Ccr5,Ccr7,Cebpb,Dhx9,Ifng,Ldlr,Lgals1,Nfkbia,Nfkbiz,Nkg7,Tnfsf11,Ttbk1</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">67</td>
<td style="text-align: right;">6.46</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">3.577709</td>
<td style="text-align: right;">1.5149549</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:1901701</td>
<td style="text-align: left;">GO:1901701</td>
<td style="text-align: left;">cellular response to oxygen-containing compound</td>
<td style="text-align: right;">5.8e-06</td>
<td style="text-align: left;">Abcc1,Actn2,Adcy6,Add1,Adipor1,Ahr,Aldh2,Aplp1,Ar,Atm,Atp2b1,Atp2b4,Baiap3,Camk2n1,Casp1,Castor1,Ccl5,Cebpb,Chrm4,Cmpk2,Cpne7,Creb1,Crk,Ctnna1,Cyfip1,Dhfr,Egr1,Ern1,Esrra,Fos,Fosb,Foxo1,Foxp1,Gna15,Gnaq,Gpr68,Gsk3b,Hmgb1,Icam1,Id2,Ide,Ifng,Inppl1,Insig1,Irs2,Itga4,Itpr3,Jun,Junb,Jund,Kdm6b,Klrk1,Lars,Ldlr,Lilrb4a,Ly6c1,Map4k1,Mmp9,Nfkbia,Nfkbiz,Nr1d1,Nr4a1,Nr4a2,Pcgf2,Pde3b,Pdgfb,Phip,Pim1,Plekha1,Plscr3,Prkar1a,Ptpn11,Ptpn22,Ptprv,Ramp3,Rhob,Rora,Rps6,Rxrb,Ryr1,Selenot,Sesn1,Setx,Sidt2,Sirt2,Slc12a2,Smarcc1,Snhg20,Socs3,Sorl1,Stk39,Syt11,Tbc1d4,Tbxa2r,Tiam1,Tnfaip3,Trerf1,Trib1,Trim12a,Ubr2,Ucp2,Uso1,Vamp2,Vim,Zbtb20,Zfp36,Zfp703</td>
<td style="text-align: right;">107</td>
<td style="text-align: right;">603</td>
<td style="text-align: right;">58.11</td>
<td style="text-align: right;">107</td>
<td style="text-align: right;">5.317051</td>
<td style="text-align: right;">0.9019446</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0001774</td>
<td style="text-align: left;">GO:0001774</td>
<td style="text-align: left;">microglial cell activation</td>
<td style="text-align: right;">8.3e-06</td>
<td style="text-align: left;">Atm,Casp1,Cst7,Ifng,Ifngr2,Jun,Ldlr,Nr1d1,Syt11,Ttbk1</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">21</td>
<td style="text-align: right;">2.02</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">1.264911</td>
<td style="text-align: right;">1.2315403</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0007186</td>
<td style="text-align: left;">GO:0007186</td>
<td style="text-align: left;">G protein-coupled receptor signaling pathway</td>
<td style="text-align: right;">1.8e-05</td>
<td style="text-align: left;">Acpp,Actn2,Ada,Adcy6,Adgre5,Akap12,Aplp1,Arhgef1,Arhgef12,Atp2b4,Baiap3,Ccl5,Ccr2,Ccr4,Ccr5,Ccr6,Ccr7,Chrm4,Cish,Cxcr3,Cxcr6,Cysltr2,Dgka,F2r,F2rl2,Gna15,Gnaq,Gng2,Gpr15,Gpr18,Gpr55,Gpr68,Gpr83,Homer3,Itpr3,Lpar5,Lpar6,Penk,Plcb2,Prkar1a,Ramp3,Rgs1,Rgs11,Rgs2,Rgs3,S1pr1,Sorl1,Tbxa2r,Vipr1</td>
<td style="text-align: right;">49</td>
<td style="text-align: right;">202</td>
<td style="text-align: right;">19.47</td>
<td style="text-align: right;">49</td>
<td style="text-align: right;">3.000000</td>
<td style="text-align: right;">0.8968280</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</section>
<section id="save-genetonic-object" class="level3">
<h3 class="anchored" data-anchor-id="save-genetonic-object">Save Genetonic object</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(subset.dds, subset.res, subset.topgo, sf.df.anno, </span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>     <span class="at">file =</span> <span class="st">"2024_01_10_results/genetonic_results/mix.4.b.VS.rej.4.b.Rdata"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="run-genetonic" class="level3">
<h3 class="anchored" data-anchor-id="run-genetonic">Run GeneTonic</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">GeneTonic</span>(<span class="at">dds =</span> subset.dds,</span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">res_de =</span> subset.res,</span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">res_enrich =</span> subset.topgo,</span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">annotation_obj =</span> sf.df.anno,</span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">project_id =</span> <span class="st">"Mixed CD4 H2Kb vs Reject CD4 H2Kb"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<hr>
</section>
</section>
</section>
<section id="subset-deseq-genetonic-function" class="level1">
<h1>Subset, DESeq, GeneTonic function</h1>
<p>Next, I wrote the function that can subset the data, perform DESeq, save the results, and build/save a GeneTonic object.</p>
<p>The only absolutely essential inputs for a function like this are the comparison conditions, called <code>con1</code> and <code>con2</code>.</p>
<p>However, the condition names lack some description and don’t have a consistent structure. I defined two now variables, <code>name1</code> and <code>name2</code>, which do a better job of communicating condition, cell type, and H2K status of each condition.</p>
<p>The code chunk below is quite large, but it essentially just the last ~10 code chunks combined with two minor changes:</p>
<ul>
<li><p><code>con1</code> and <code>con2</code> take the place of <code>Mixed CD4 H2Kb</code> and <code>Reject CD4 H2Kb</code>.</p></li>
<li><p>I used <code>paste0()</code> in combination with <code>name1</code> and <code>name2</code> to name the result files.</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>condition_deseq <span class="ot">&lt;-</span> <span class="cf">function</span>(con1, con2, name1, name2) {</span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a vector that has the samples we want to compare</span></span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>  subset.vector <span class="ot">&lt;-</span> meta <span class="sc">|&gt;</span> </span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(condition <span class="sc">==</span> con1 <span class="sc">|</span> condition <span class="sc">==</span> con2) <span class="sc">|&gt;</span> </span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(folder_name)</span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset the sf.df</span></span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>  subset.sf.df <span class="ot">&lt;-</span> sf.df <span class="sc">|&gt;</span> </span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(subset.vector)) </span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-12"><a href="#cb90-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter out low counts</span></span>
<span id="cb90-13"><a href="#cb90-13" aria-hidden="true" tabindex="-1"></a>  subset.sf.df <span class="ot">&lt;-</span> subset.sf.df <span class="sc">|&gt;</span> </span>
<span id="cb90-14"><a href="#cb90-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">row.sum =</span> <span class="fu">apply</span>(subset.sf.df, <span class="dv">1</span>, <span class="cf">function</span>(x) <span class="fu">sum</span>(x <span class="sc">&gt;</span> <span class="dv">10</span>))) <span class="sc">|&gt;</span> </span>
<span id="cb90-15"><a href="#cb90-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># All values must be more than 10 (set to x if only x value is satisfied)</span></span>
<span id="cb90-16"><a href="#cb90-16" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(row.sum <span class="sc">==</span> <span class="fu">length</span>(subset.vector)) <span class="sc">|&gt;</span> </span>
<span id="cb90-17"><a href="#cb90-17" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Remove row.sum, it is not needing moving forward</span></span>
<span id="cb90-18"><a href="#cb90-18" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>row.sum)</span>
<span id="cb90-19"><a href="#cb90-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-20"><a href="#cb90-20" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filtering metadata so only relevant samples are put into the DESeq object</span></span>
<span id="cb90-21"><a href="#cb90-21" aria-hidden="true" tabindex="-1"></a>  subset_meta <span class="ot">&lt;-</span> meta <span class="sc">|&gt;</span> </span>
<span id="cb90-22"><a href="#cb90-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(folder_name <span class="sc">%in%</span> subset.vector)</span>
<span id="cb90-23"><a href="#cb90-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-24"><a href="#cb90-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create subset salmon matrix</span></span>
<span id="cb90-25"><a href="#cb90-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-26"><a href="#cb90-26" aria-hidden="true" tabindex="-1"></a>  subset.salmon.matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(subset.sf.df)</span>
<span id="cb90-27"><a href="#cb90-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-28"><a href="#cb90-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a dataframe of subset samples and their condition, need to created SE object</span></span>
<span id="cb90-29"><a href="#cb90-29" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-30"><a href="#cb90-30" aria-hidden="true" tabindex="-1"></a>  subset.coldata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample =</span> subset_meta<span class="sc">$</span>folder_name,</span>
<span id="cb90-31"><a href="#cb90-31" aria-hidden="true" tabindex="-1"></a>                        <span class="at">condition =</span> <span class="fu">as.factor</span>(subset_meta<span class="sc">$</span>condition),</span>
<span id="cb90-32"><a href="#cb90-32" aria-hidden="true" tabindex="-1"></a>                        <span class="at">row.names =</span> <span class="st">"sample"</span>)</span>
<span id="cb90-33"><a href="#cb90-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create SE object</span></span>
<span id="cb90-34"><a href="#cb90-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-35"><a href="#cb90-35" aria-hidden="true" tabindex="-1"></a>  subset.se <span class="ot">&lt;-</span> <span class="fu">SummarizedExperiment</span>(<span class="at">assays =</span> <span class="fu">list</span>(<span class="at">counts =</span> subset.salmon.matrix),</span>
<span id="cb90-36"><a href="#cb90-36" aria-hidden="true" tabindex="-1"></a>                             <span class="at">colData =</span> subset.coldata)</span>
<span id="cb90-37"><a href="#cb90-37" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-38"><a href="#cb90-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create DDS object</span></span>
<span id="cb90-39"><a href="#cb90-39" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-40"><a href="#cb90-40" aria-hidden="true" tabindex="-1"></a>  subset.dds <span class="ot">=</span> <span class="fu">DESeqDataSet</span>(subset.se, <span class="at">design =</span> <span class="sc">~</span> condition)</span>
<span id="cb90-41"><a href="#cb90-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-42"><a href="#cb90-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># DESeq and results looking at our subsetted samples</span></span>
<span id="cb90-43"><a href="#cb90-43" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-44"><a href="#cb90-44" aria-hidden="true" tabindex="-1"></a>  subset.dds <span class="ot">=</span> <span class="fu">DESeq</span>(subset.dds)</span>
<span id="cb90-45"><a href="#cb90-45" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-46"><a href="#cb90-46" aria-hidden="true" tabindex="-1"></a>  subset.res <span class="ot">&lt;-</span> <span class="fu">results</span>(subset.dds,</span>
<span id="cb90-47"><a href="#cb90-47" aria-hidden="true" tabindex="-1"></a>                          <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"condition"</span>, con1, con2),</span>
<span id="cb90-48"><a href="#cb90-48" aria-hidden="true" tabindex="-1"></a>                          <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb90-49"><a href="#cb90-49" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-50"><a href="#cb90-50" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create annotated results, join it to raw counts, and filter out NA genes</span></span>
<span id="cb90-51"><a href="#cb90-51" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-52"><a href="#cb90-52" aria-hidden="true" tabindex="-1"></a>  subset.ids <span class="ot">&lt;-</span> subset.sf.df <span class="sc">|&gt;</span> </span>
<span id="cb90-53"><a href="#cb90-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"gene_id"</span>)</span>
<span id="cb90-54"><a href="#cb90-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-55"><a href="#cb90-55" aria-hidden="true" tabindex="-1"></a>  subset.anno <span class="ot">&lt;-</span> <span class="fu">deseqresult2df</span>(subset.res) <span class="sc">|&gt;</span> </span>
<span id="cb90-56"><a href="#cb90-56" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene_id =</span> id) <span class="sc">|&gt;</span> </span>
<span id="cb90-57"><a href="#cb90-57" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">gene_symbol =</span> <span class="fu">mapIds</span>(org.Mm.eg.db, </span>
<span id="cb90-58"><a href="#cb90-58" aria-hidden="true" tabindex="-1"></a>                                <span class="at">keys =</span> gene_id, </span>
<span id="cb90-59"><a href="#cb90-59" aria-hidden="true" tabindex="-1"></a>                                <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>, </span>
<span id="cb90-60"><a href="#cb90-60" aria-hidden="true" tabindex="-1"></a>                                <span class="at">column =</span> <span class="st">"SYMBOL"</span>, </span>
<span id="cb90-61"><a href="#cb90-61" aria-hidden="true" tabindex="-1"></a>                                <span class="at">multiVals =</span> <span class="st">"first"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb90-62"><a href="#cb90-62" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">gene_entrez =</span> <span class="fu">mapIds</span>(org.Mm.eg.db, <span class="at">keys =</span> gene_id, </span>
<span id="cb90-63"><a href="#cb90-63" aria-hidden="true" tabindex="-1"></a>                                <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>, </span>
<span id="cb90-64"><a href="#cb90-64" aria-hidden="true" tabindex="-1"></a>                                <span class="at">column =</span> <span class="st">"ENTREZID"</span>, </span>
<span id="cb90-65"><a href="#cb90-65" aria-hidden="true" tabindex="-1"></a>                                <span class="at">multiVals =</span> <span class="st">"first"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb90-66"><a href="#cb90-66" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">gene_desc =</span> <span class="fu">mapIds</span>(org.Mm.eg.db, </span>
<span id="cb90-67"><a href="#cb90-67" aria-hidden="true" tabindex="-1"></a>                              <span class="at">keys =</span> gene_id, </span>
<span id="cb90-68"><a href="#cb90-68" aria-hidden="true" tabindex="-1"></a>                              <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>, </span>
<span id="cb90-69"><a href="#cb90-69" aria-hidden="true" tabindex="-1"></a>                              <span class="at">column =</span> <span class="st">"GENENAME"</span>, </span>
<span id="cb90-70"><a href="#cb90-70" aria-hidden="true" tabindex="-1"></a>                              <span class="at">multiVals =</span> <span class="st">"first"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb90-71"><a href="#cb90-71" aria-hidden="true" tabindex="-1"></a>    <span class="fu">relocate</span>(gene_symbol, gene_entrez, gene_desc, <span class="at">.after =</span> gene_id) <span class="sc">|&gt;</span> </span>
<span id="cb90-72"><a href="#cb90-72" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(subset.ids, </span>
<span id="cb90-73"><a href="#cb90-73" aria-hidden="true" tabindex="-1"></a>              <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"gene_id"</span><span class="ot">=</span><span class="st">"gene_id"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb90-74"><a href="#cb90-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">na.omit</span>()</span>
<span id="cb90-75"><a href="#cb90-75" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-76"><a href="#cb90-76" aria-hidden="true" tabindex="-1"></a>  <span class="co"># write csv file</span></span>
<span id="cb90-77"><a href="#cb90-77" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-78"><a href="#cb90-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">write_csv</span>(subset.anno, <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"2024_01_10_results/DESeq_results/"</span>, name1, <span class="st">".VS."</span>, name2, <span class="st">".csv"</span>))</span>
<span id="cb90-79"><a href="#cb90-79" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-80"><a href="#cb90-80" aria-hidden="true" tabindex="-1"></a>  <span class="co"># GeneTonic Prep</span></span>
<span id="cb90-81"><a href="#cb90-81" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-82"><a href="#cb90-82" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Enriched results object</span></span>
<span id="cb90-83"><a href="#cb90-83" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-84"><a href="#cb90-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">deseqresult2df</span>(subset.res) <span class="sc">|&gt;</span> </span>
<span id="cb90-85"><a href="#cb90-85" aria-hidden="true" tabindex="-1"></a>    <span class="co"># rownames_to_column(var = "Ensembl") |&gt; </span></span>
<span id="cb90-86"><a href="#cb90-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Symbol =</span> <span class="fu">mapIds</span>(org.Mm.eg.db, </span>
<span id="cb90-87"><a href="#cb90-87" aria-hidden="true" tabindex="-1"></a>                           <span class="at">keys =</span> id, </span>
<span id="cb90-88"><a href="#cb90-88" aria-hidden="true" tabindex="-1"></a>                           <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>, </span>
<span id="cb90-89"><a href="#cb90-89" aria-hidden="true" tabindex="-1"></a>                           <span class="at">column =</span> <span class="st">"SYMBOL"</span>, </span>
<span id="cb90-90"><a href="#cb90-90" aria-hidden="true" tabindex="-1"></a>                           <span class="at">multiVals =</span> <span class="st">"first"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb90-91"><a href="#cb90-91" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span> </span>
<span id="cb90-92"><a href="#cb90-92" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(Symbol) <span class="ot">-&gt;</span> enrich.of.interest</span>
<span id="cb90-93"><a href="#cb90-93" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-94"><a href="#cb90-94" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Background gene symbols</span></span>
<span id="cb90-95"><a href="#cb90-95" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-96"><a href="#cb90-96" aria-hidden="true" tabindex="-1"></a>  bg_ids <span class="ot">&lt;-</span> <span class="fu">deseqresult2df</span>(subset.res) <span class="sc">|&gt;</span> </span>
<span id="cb90-97"><a href="#cb90-97" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Symbol =</span> <span class="fu">mapIds</span>(org.Mm.eg.db, </span>
<span id="cb90-98"><a href="#cb90-98" aria-hidden="true" tabindex="-1"></a>                           <span class="at">keys =</span> id, </span>
<span id="cb90-99"><a href="#cb90-99" aria-hidden="true" tabindex="-1"></a>                           <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>, </span>
<span id="cb90-100"><a href="#cb90-100" aria-hidden="true" tabindex="-1"></a>                           <span class="at">column =</span> <span class="st">"SYMBOL"</span>, </span>
<span id="cb90-101"><a href="#cb90-101" aria-hidden="true" tabindex="-1"></a>                           <span class="at">multiVals =</span> <span class="st">"first"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb90-102"><a href="#cb90-102" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(Symbol)</span>
<span id="cb90-103"><a href="#cb90-103" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-104"><a href="#cb90-104" aria-hidden="true" tabindex="-1"></a>  <span class="do">### Gene annotation object</span></span>
<span id="cb90-105"><a href="#cb90-105" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-106"><a href="#cb90-106" aria-hidden="true" tabindex="-1"></a>  sf.df.anno <span class="ot">&lt;-</span> <span class="fu">deseqresult2df</span>(subset.res) <span class="sc">|&gt;</span> </span>
<span id="cb90-107"><a href="#cb90-107" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">gene_id =</span> id) <span class="sc">|&gt;</span> </span>
<span id="cb90-108"><a href="#cb90-108" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">gene_name =</span> <span class="fu">mapIds</span>(org.Mm.eg.db, </span>
<span id="cb90-109"><a href="#cb90-109" aria-hidden="true" tabindex="-1"></a>                              <span class="at">keys =</span> gene_id, </span>
<span id="cb90-110"><a href="#cb90-110" aria-hidden="true" tabindex="-1"></a>                              <span class="at">keytype =</span> <span class="st">"ENSEMBL"</span>, </span>
<span id="cb90-111"><a href="#cb90-111" aria-hidden="true" tabindex="-1"></a>                              <span class="at">column =</span> <span class="st">"SYMBOL"</span>, </span>
<span id="cb90-112"><a href="#cb90-112" aria-hidden="true" tabindex="-1"></a>                              <span class="at">multiVals =</span> <span class="st">"first"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb90-113"><a href="#cb90-113" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(gene_id, gene_name)</span>
<span id="cb90-114"><a href="#cb90-114" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-115"><a href="#cb90-115" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Perform Pathway Analysis</span></span>
<span id="cb90-116"><a href="#cb90-116" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-117"><a href="#cb90-117" aria-hidden="true" tabindex="-1"></a>  subset.topgo <span class="ot">=</span> pcaExplorer<span class="sc">::</span><span class="fu">topGOtable</span>(enrich.of.interest,</span>
<span id="cb90-118"><a href="#cb90-118" aria-hidden="true" tabindex="-1"></a>                                         bg_ids,</span>
<span id="cb90-119"><a href="#cb90-119" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">ontology =</span> <span class="st">"BP"</span>,</span>
<span id="cb90-120"><a href="#cb90-120" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">mapping =</span> <span class="st">"org.Mm.eg.db"</span>,</span>
<span id="cb90-121"><a href="#cb90-121" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">geneID =</span> <span class="st">"symbol"</span>)</span>
<span id="cb90-122"><a href="#cb90-122" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-123"><a href="#cb90-123" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Reshape for Genetonic</span></span>
<span id="cb90-124"><a href="#cb90-124" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-125"><a href="#cb90-125" aria-hidden="true" tabindex="-1"></a>  subset.topgo <span class="ot">&lt;-</span> <span class="fu">shake_topGOtableResult</span>(subset.topgo)</span>
<span id="cb90-126"><a href="#cb90-126" aria-hidden="true" tabindex="-1"></a>  subset.topgo <span class="ot">&lt;-</span> <span class="fu">get_aggrscores</span>(subset.topgo, subset.res, sf.df.anno)</span>
<span id="cb90-127"><a href="#cb90-127" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-128"><a href="#cb90-128" aria-hidden="true" tabindex="-1"></a>  <span class="co"># save info for GeneTonic</span></span>
<span id="cb90-129"><a href="#cb90-129" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb90-130"><a href="#cb90-130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">save</span>(subset.dds, subset.res, subset.topgo, sf.df.anno, <span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"2024_01_10_results/genetonic_results/"</span>, name1, <span class="st">".VS."</span>, name2, <span class="st">".Rdata"</span>))</span>
<span id="cb90-131"><a href="#cb90-131" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="test-run-of-the-function-using-mixed-cd4-h2kb-cells-vs-the-other-3-h2kb-conditions" class="level2">
<h2 class="anchored" data-anchor-id="test-run-of-the-function-using-mixed-cd4-h2kb-cells-vs-the-other-3-h2kb-conditions">Test run of the function using Mixed CD4 H2Kb cells vs the other 3 H2Kb conditions:</h2>
<p>Some words here about using pmap vs lapply, mapply, etc…</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>con1 <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Mixed CD4 HLA B"</span>, <span class="st">"Mixed CD4 HLA B"</span>, <span class="st">"Mixed CD4 HLA B"</span>)</span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>con2 <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Reject CD4"</span>, <span class="st">"Autol C57 CD4"</span>, <span class="st">"WT control CD4"</span>)</span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>name1 <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"mix.4.b"</span>, <span class="st">"mix.4.b"</span>, <span class="st">"mix.4.b"</span>)</span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a>name2 <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"rej.4.b"</span>, <span class="st">"auto.4.b"</span>, <span class="st">"wt.con.4.b"</span>)</span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-6"><a href="#cb91-6" aria-hidden="true" tabindex="-1"></a>mix.cd4.b.vs.other.b.list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">list</span>(con1), <span class="fu">list</span>(con2), <span class="fu">list</span>(name1), <span class="fu">list</span>(name2))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pmap</span>(mix.cd4.b.vs.other.b.list, condition_deseq)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This took some time to make, but likely saved me time in this comparison. Not to mention future data.</p>
<p>The initial comparisons used <code>condition</code> as the main variable. But with a small amount of changes to the metadata and the function, it could run on any column in the metadata.</p>
<p>For example, if I want to combine the mouse strain data (CD4 T cells from both mixed groups, or both auto groups), I could:</p>
<ul>
<li><p>Create a new column in the metadata called <code>group</code> using <code>mutate()</code> and <code>case_when()</code> to define my <code>group</code> column.</p></li>
<li><p>Change each <code>condition</code> in the function to <code>group</code></p></li>
<li><p>Make a new input list of <code>con1</code>, <code>con2</code>, <code>name1</code>, and <code>name2</code></p></li>
<li><p>Run!</p></li>
</ul>
</section>
</section>
<section id="section" class="level1">
<h1>————————————————————————</h1>
<p>Next!</p>
</section>
<section id="generating-heatmaps-volcano-plots-and-genetonic-plots" class="level1">
<h1>Generating heatmaps, volcano plots, and GeneTonic plots</h1>
<section id="single-comparison-walk-through" class="level2">
<h2 class="anchored" data-anchor-id="single-comparison-walk-through">Single comparison walk through</h2>
<p>Now that we have our DESeq data and GeneTonic object data, I made some plots for each comparison. Four plots for each: - Heatmap of top 25 DE genes by adjusted p value - Heatmap of top 25 DE genes by absolute log2FC - Volcano plot - GeneTonic Pathway Analysis plot (The call it “Summary Overview”)</p>
<p>I could have created a folder for heatmaps, one for volcano plots, etc…</p>
<p>But that is hard to navigate, with 14 plots side by side. I thought it would be better for each comparison to have its own folder that contains the relevant plots.</p>
<p>We will need to read in the data for the plots. We technically have it saved in our variables for this notebook, but doing it this way helps us fix any issues when applying the process to the rest of the comparisons.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"2024_01_10_results/genetonic_results/mix.4.b.VS.rej.4.b.Rdata"</span>))</span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a>deseq.df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"2024_01_10_results/deseq_results/mix.4.b.VS.rej.4.b.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>New results folder, which will contain the subfolders for each comparison.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>folder.name <span class="ot">&lt;-</span> <span class="st">"2024_01_19_results"</span></span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">exists</span>(folder.name)){</span>
<span id="cb94-4"><a href="#cb94-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dir.create</span>(folder.name)</span>
<span id="cb94-5"><a href="#cb94-5" aria-hidden="true" tabindex="-1"></a>  }</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(folder.name): '2024_01_19_results' already exists</code></pre>
</div>
</div>
<section id="heatmap-generation" class="level3">
<h3 class="anchored" data-anchor-id="heatmap-generation">Heatmap generation</h3>
<p>The sample names in the counts matrix are typically displayed under each column of the heatmap. But I found they made the plot harder to interpret, so I made an annotation dataframe to make them more readable.</p>
<p>Another sticking point: My DESeq results have the subset counts for my heatmaps, but there is no consistent naming structure. Our samples all have <code>L002</code>, but the public data do not. Some comparisons have 6 samples, others only 5.</p>
<p>I found the best way to isolate the counts was to remove the common rows they all share: the DESeq results at the front of the dataframe, with <code>dplyr::select(deseq.df, -1:-10)</code></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This removes all the columns that are not raw counts from the DESeq results</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>sample.df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(deseq.df, <span class="sc">-</span><span class="dv">1</span><span class="sc">:-</span><span class="dv">10</span>)</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter metadata for one of the condition comparisons</span></span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a>subset1 <span class="ot">&lt;-</span> meta <span class="sc">|&gt;</span> </span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(condition <span class="sc">==</span> <span class="st">"Mixed CD4 HLA B"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(folder_name)</span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a><span class="co"># change - to . to match column name format</span></span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>subset1 <span class="ot">&lt;-</span> subset1 <span class="sc">|&gt;</span> </span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace_all</span>(<span class="st">"-"</span>, <span class="st">"."</span>)</span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a><span class="co"># R adds X to all column names that start with a number, so we need to add X as well</span></span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a>subset1 <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"X"</span>, subset1)</span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the annotation dataframe where the column name is matched with its condition</span></span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a>anno <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample_name =</span> <span class="fu">colnames</span>(sample.df)) <span class="sc">|&gt;</span> </span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">condition =</span> <span class="fu">ifelse</span>(sample_name <span class="sc">%in%</span> subset1, <span class="st">"Mixed CD4 HLA B"</span>, <span class="st">"Reject CD4"</span>)) <span class="sc">|&gt;</span> </span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">"sample_name"</span>)</span>
<span id="cb96-17"><a href="#cb96-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-18"><a href="#cb96-18" aria-hidden="true" tabindex="-1"></a><span class="co"># defining colors for the annotation</span></span>
<span id="cb96-19"><a href="#cb96-19" aria-hidden="true" tabindex="-1"></a>condition <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"orange"</span>, <span class="st">"blue"</span>)</span>
<span id="cb96-20"><a href="#cb96-20" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(condition) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Mixed CD4 HLA B"</span>, <span class="st">"Reject CD4"</span>)</span>
<span id="cb96-21"><a href="#cb96-21" aria-hidden="true" tabindex="-1"></a>anno_colors <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">condition =</span> condition)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>anno</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="kable-table">
<table class="table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">condition</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">X52.CD4_S37_L002</td>
<td style="text-align: left;">Reject CD4</td>
</tr>
<tr class="even">
<td style="text-align: left;">X53.CD4_S39_L002</td>
<td style="text-align: left;">Reject CD4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">X57.CD4.B_S45_L002</td>
<td style="text-align: left;">Mixed CD4 HLA B</td>
</tr>
<tr class="even">
<td style="text-align: left;">X58.CD4.B_S49_L002</td>
<td style="text-align: left;">Mixed CD4 HLA B</td>
</tr>
<tr class="odd">
<td style="text-align: left;">X60.CD4.B_S53_L002</td>
<td style="text-align: left;">Mixed CD4 HLA B</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>anno_colors</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$condition
Mixed CD4 HLA B      Reject CD4 
       "orange"          "blue" </code></pre>
</div>
</div>
</section>
<section id="create-and-save-heatmaps" class="level3">
<h3 class="anchored" data-anchor-id="create-and-save-heatmaps">Create and save heatmaps</h3>
<p>Heatmap for the top 25 padj</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>deseq.df <span class="sc">|&gt;</span> </span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">25</span>) <span class="sc">|&gt;</span> </span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"gene_symbol"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="dv">1</span><span class="sc">:-</span><span class="dv">9</span>) <span class="sc">|&gt;</span> </span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pheatmap</span>(<span class="at">scale =</span> <span class="st">"row"</span>,</span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">12</span>,</span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">11</span>,</span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb100-9"><a href="#cb100-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">treeheight_col =</span> <span class="dv">0</span>,</span>
<span id="cb100-10"><a href="#cb100-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation =</span> anno,</span>
<span id="cb100-11"><a href="#cb100-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_colors =</span> anno_colors,</span>
<span id="cb100-12"><a href="#cb100-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"Top 25 genes by p-value"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb100-13"><a href="#cb100-13" aria-hidden="true" tabindex="-1"></a>         <span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">"2024_01_19_results/mix.4.b.VS.rej.4.b/mix.4.b.VS.rej.4.b.padj.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>deseq.df <span class="sc">|&gt;</span> </span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span> </span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">abs</span>(<span class="fu">c</span>(log2FoldChange)))) <span class="sc">|&gt;</span> </span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">25</span>) <span class="sc">|&gt;</span> </span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"gene_symbol"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="dv">1</span><span class="sc">:-</span><span class="dv">9</span>) <span class="sc">|&gt;</span></span>
<span id="cb102-7"><a href="#cb102-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pheatmap</span>(<span class="at">scale =</span> <span class="st">"row"</span>,</span>
<span id="cb102-8"><a href="#cb102-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">12</span>,</span>
<span id="cb102-9"><a href="#cb102-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">11</span>,</span>
<span id="cb102-10"><a href="#cb102-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb102-11"><a href="#cb102-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">treeheight_col =</span> <span class="dv">0</span>,</span>
<span id="cb102-12"><a href="#cb102-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation =</span> anno,</span>
<span id="cb102-13"><a href="#cb102-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">annotation_colors =</span> anno_colors,</span>
<span id="cb102-14"><a href="#cb102-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="st">"Top 25 genes by absolute log2FC"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb102-15"><a href="#cb102-15" aria-hidden="true" tabindex="-1"></a>         <span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">"2024_01_19_results/mix.4.b.VS.rej.4.b/mix.4.b.VS.rej.4.b.log2FC.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="volcano-plot-of-de-genes" class="level2">
<h2 class="anchored" data-anchor-id="volcano-plot-of-de-genes">Volcano plot of DE genes</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>volc.plot <span class="ot">&lt;-</span> <span class="fu">EnhancedVolcano</span>(<span class="at">toptable =</span> deseq.df,</span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">lab =</span> deseq.df<span class="sc">$</span>gene_symbol,</span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">x =</span> <span class="st">"log2FoldChange"</span>,</span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">y =</span> <span class="st">"padj"</span>,</span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a>                <span class="co"># xlim = c(-3, 3),</span></span>
<span id="cb104-6"><a href="#cb104-6" aria-hidden="true" tabindex="-1"></a>                <span class="co"># ylim = c(-0.5, 8),</span></span>
<span id="cb104-7"><a href="#cb104-7" aria-hidden="true" tabindex="-1"></a>                <span class="at">title =</span> <span class="st">"Mixed CD4 HLA B vs Reject CD4"</span>,</span>
<span id="cb104-8"><a href="#cb104-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">subtitle =</span> <span class="st">"DESeq2 Results"</span>,</span>
<span id="cb104-9"><a href="#cb104-9" aria-hidden="true" tabindex="-1"></a>                <span class="co"># FCcutoff = ,</span></span>
<span id="cb104-10"><a href="#cb104-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">pCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb104-11"><a href="#cb104-11" aria-hidden="true" tabindex="-1"></a>                <span class="co"># #labFace = "bold",</span></span>
<span id="cb104-12"><a href="#cb104-12" aria-hidden="true" tabindex="-1"></a>                <span class="at">legendPosition =</span> <span class="st">"none"</span>,</span>
<span id="cb104-13"><a href="#cb104-13" aria-hidden="true" tabindex="-1"></a>                <span class="at">caption =</span> <span class="st">""</span>) </span>
<span id="cb104-14"><a href="#cb104-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-15"><a href="#cb104-15" aria-hidden="true" tabindex="-1"></a>volc.plot <span class="sc">|&gt;</span> </span>
<span id="cb104-16"><a href="#cb104-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">"2024_01_19_results/mix.4.b.VS.rej.4.b/mix.4.b.VS.rej.4.b.volcano.png"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a>volc.plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="genetonic-pathway-summary-plot" class="level2">
<h2 class="anchored" data-anchor-id="genetonic-pathway-summary-plot">GeneTonic Pathway summary plot</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">gs_summary_overview</span>(<span class="at">res_enrich =</span> subset.topgo,</span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">n_gs =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb107-3"><a href="#cb107-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Pathway Enrichment"</span>,</span>
<span id="cb107-4"><a href="#cb107-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">"Mixed CD4 HLA B vs Reject CD4"</span>)</span>
<span id="cb107-5"><a href="#cb107-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-6"><a href="#cb107-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="st">"2024_01_19_results/mix.4.b.VS.rej.4.b/mix.4.b.VS.rej.4.b.summary.png"</span>,</span>
<span id="cb107-7"><a href="#cb107-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">plot =</span> p,</span>
<span id="cb107-8"><a href="#cb107-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">width =</span> <span class="dv">11</span>,</span>
<span id="cb107-9"><a href="#cb107-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">height =</span> <span class="dv">8</span>,</span>
<span id="cb107-10"><a href="#cb107-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb107-11"><a href="#cb107-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb107-12"><a href="#cb107-12" aria-hidden="true" tabindex="-1"></a>p</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="plot-function" class="level1">
<h1>Plot function</h1>
<p>Now that we’ve generated each of our plots, we can create the function that does it all for us. And we have set it up similar to the DESeq function. All we need to provide is the conditions for comparison as con1 and con2 and the name abbreviations as name1 and name2 again.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>plot_results <span class="ot">&lt;-</span> <span class="cf">function</span>(con1, con2, name1, name2){</span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">load</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"2024_01_10_results/genetonic_results/"</span>, name1, <span class="st">".VS."</span>, name2, <span class="st">".Rdata"</span>))</span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a>  deseq.df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="fu">paste0</span>(<span class="st">"2024_01_10_results/deseq_results/"</span>, name1, <span class="st">".VS."</span>, name2, <span class="st">".csv"</span>)) </span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-6"><a href="#cb108-6" aria-hidden="true" tabindex="-1"></a>  results.folder <span class="ot">&lt;-</span> <span class="fu">paste0</span>(folder.name, <span class="st">"/"</span>, name1, <span class="st">".VS."</span>, name2, <span class="st">"/"</span>)</span>
<span id="cb108-7"><a href="#cb108-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-8"><a href="#cb108-8" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">exists</span>(results.folder)){</span>
<span id="cb108-9"><a href="#cb108-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dir.create</span>(results.folder)}</span>
<span id="cb108-10"><a href="#cb108-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-11"><a href="#cb108-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter the deseq.df so we can pull out the colnames for the annotation df</span></span>
<span id="cb108-12"><a href="#cb108-12" aria-hidden="true" tabindex="-1"></a>  sample.df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(deseq.df, <span class="sc">-</span><span class="dv">1</span><span class="sc">:-</span><span class="dv">10</span>)</span>
<span id="cb108-13"><a href="#cb108-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-14"><a href="#cb108-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter metadata for one of the condition comparisons</span></span>
<span id="cb108-15"><a href="#cb108-15" aria-hidden="true" tabindex="-1"></a>  subset1 <span class="ot">&lt;-</span> meta <span class="sc">|&gt;</span> </span>
<span id="cb108-16"><a href="#cb108-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(condition <span class="sc">==</span> con1) <span class="sc">|&gt;</span> </span>
<span id="cb108-17"><a href="#cb108-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pull</span>(folder_name)</span>
<span id="cb108-18"><a href="#cb108-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># change - to . to match column name format</span></span>
<span id="cb108-19"><a href="#cb108-19" aria-hidden="true" tabindex="-1"></a>  subset1 <span class="ot">&lt;-</span> subset1 <span class="sc">|&gt;</span> </span>
<span id="cb108-20"><a href="#cb108-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">str_replace_all</span>(<span class="st">"-"</span>, <span class="st">"."</span>)</span>
<span id="cb108-21"><a href="#cb108-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># R adds X to all column names that start with a number, so we need to add X as well</span></span>
<span id="cb108-22"><a href="#cb108-22" aria-hidden="true" tabindex="-1"></a>  subset1 <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"X"</span>, subset1)</span>
<span id="cb108-23"><a href="#cb108-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create the annotation dataframe where the column name is matched with its condition</span></span>
<span id="cb108-24"><a href="#cb108-24" aria-hidden="true" tabindex="-1"></a>  anno <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample_name =</span> <span class="fu">colnames</span>(sample.df)) <span class="sc">|&gt;</span> </span>
<span id="cb108-25"><a href="#cb108-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">condition =</span> <span class="fu">ifelse</span>(sample_name <span class="sc">%in%</span> subset1, con1, con2)) <span class="sc">|&gt;</span> </span>
<span id="cb108-26"><a href="#cb108-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">column_to_rownames</span>(<span class="st">"sample_name"</span>)</span>
<span id="cb108-27"><a href="#cb108-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-28"><a href="#cb108-28" aria-hidden="true" tabindex="-1"></a>  <span class="co"># defining colors for the annotation</span></span>
<span id="cb108-29"><a href="#cb108-29" aria-hidden="true" tabindex="-1"></a>  condition <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"orange"</span>, <span class="st">"blue"</span>)</span>
<span id="cb108-30"><a href="#cb108-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(condition) <span class="ot">&lt;-</span> <span class="fu">c</span>(con1, con2)</span>
<span id="cb108-31"><a href="#cb108-31" aria-hidden="true" tabindex="-1"></a>  anno_colors <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">condition =</span> condition)</span>
<span id="cb108-32"><a href="#cb108-32" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-33"><a href="#cb108-33" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Top 25 DE by padj heatmap</span></span>
<span id="cb108-34"><a href="#cb108-34" aria-hidden="true" tabindex="-1"></a>  deseq.df <span class="sc">|&gt;</span> </span>
<span id="cb108-35"><a href="#cb108-35" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">25</span>) <span class="sc">|&gt;</span> </span>
<span id="cb108-36"><a href="#cb108-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"gene_symbol"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb108-37"><a href="#cb108-37" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="dv">1</span><span class="sc">:-</span><span class="dv">9</span>) <span class="sc">|&gt;</span> </span>
<span id="cb108-38"><a href="#cb108-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pheatmap</span>(<span class="at">scale =</span> <span class="st">"row"</span>,</span>
<span id="cb108-39"><a href="#cb108-39" aria-hidden="true" tabindex="-1"></a>           <span class="at">fontsize =</span> <span class="dv">12</span>,</span>
<span id="cb108-40"><a href="#cb108-40" aria-hidden="true" tabindex="-1"></a>           <span class="at">fontsize_row =</span> <span class="dv">11</span>,</span>
<span id="cb108-41"><a href="#cb108-41" aria-hidden="true" tabindex="-1"></a>           <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb108-42"><a href="#cb108-42" aria-hidden="true" tabindex="-1"></a>           <span class="at">treeheight_col =</span> <span class="dv">0</span>,</span>
<span id="cb108-43"><a href="#cb108-43" aria-hidden="true" tabindex="-1"></a>           <span class="at">annotation =</span> anno,</span>
<span id="cb108-44"><a href="#cb108-44" aria-hidden="true" tabindex="-1"></a>           <span class="at">annotation_colors =</span> anno_colors,</span>
<span id="cb108-45"><a href="#cb108-45" aria-hidden="true" tabindex="-1"></a>           <span class="at">main =</span> <span class="st">"Top 25 genes by p-value"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb108-46"><a href="#cb108-46" aria-hidden="true" tabindex="-1"></a>           <span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="fu">paste0</span>(results.folder, name1, <span class="st">".VS."</span>, name2, <span class="st">".padj.png"</span>))</span>
<span id="cb108-47"><a href="#cb108-47" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-48"><a href="#cb108-48" aria-hidden="true" tabindex="-1"></a>  <span class="co">#p value heatmap</span></span>
<span id="cb108-49"><a href="#cb108-49" aria-hidden="true" tabindex="-1"></a>  deseq.df <span class="sc">|&gt;</span> </span>
<span id="cb108-50"><a href="#cb108-50" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">|&gt;</span> </span>
<span id="cb108-51"><a href="#cb108-51" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(<span class="fu">abs</span>(<span class="fu">c</span>(log2FoldChange)))) <span class="sc">|&gt;</span> </span>
<span id="cb108-52"><a href="#cb108-52" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">25</span>) <span class="sc">|&gt;</span> </span>
<span id="cb108-53"><a href="#cb108-53" aria-hidden="true" tabindex="-1"></a>    <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">"gene_symbol"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb108-54"><a href="#cb108-54" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="dv">1</span><span class="sc">:-</span><span class="dv">9</span>) <span class="sc">|&gt;</span> </span>
<span id="cb108-55"><a href="#cb108-55" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pheatmap</span>(<span class="at">scale =</span> <span class="st">"row"</span>,</span>
<span id="cb108-56"><a href="#cb108-56" aria-hidden="true" tabindex="-1"></a>           <span class="at">fontsize =</span> <span class="dv">12</span>,</span>
<span id="cb108-57"><a href="#cb108-57" aria-hidden="true" tabindex="-1"></a>           <span class="at">fontsize_row =</span> <span class="dv">11</span>,</span>
<span id="cb108-58"><a href="#cb108-58" aria-hidden="true" tabindex="-1"></a>           <span class="at">show_colnames =</span> <span class="cn">FALSE</span>,</span>
<span id="cb108-59"><a href="#cb108-59" aria-hidden="true" tabindex="-1"></a>           <span class="at">treeheight_col =</span> <span class="dv">0</span>,</span>
<span id="cb108-60"><a href="#cb108-60" aria-hidden="true" tabindex="-1"></a>           <span class="at">annotation =</span> anno,</span>
<span id="cb108-61"><a href="#cb108-61" aria-hidden="true" tabindex="-1"></a>           <span class="at">annotation_colors =</span> anno_colors,</span>
<span id="cb108-62"><a href="#cb108-62" aria-hidden="true" tabindex="-1"></a>           <span class="at">main =</span> <span class="st">"Top 25 genes by absolute log2FC"</span>) <span class="sc">|&gt;</span> </span>
<span id="cb108-63"><a href="#cb108-63" aria-hidden="true" tabindex="-1"></a>           <span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="fu">paste0</span>(results.folder, name1, <span class="st">".VS."</span>, name2, <span class="st">".log2FC.png"</span>))</span>
<span id="cb108-64"><a href="#cb108-64" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-65"><a href="#cb108-65" aria-hidden="true" tabindex="-1"></a>  <span class="co">#log2FC heatmap</span></span>
<span id="cb108-66"><a href="#cb108-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">EnhancedVolcano</span>(<span class="at">toptable =</span> deseq.df,</span>
<span id="cb108-67"><a href="#cb108-67" aria-hidden="true" tabindex="-1"></a>                  <span class="at">lab =</span> deseq.df<span class="sc">$</span>gene_symbol,</span>
<span id="cb108-68"><a href="#cb108-68" aria-hidden="true" tabindex="-1"></a>                  <span class="at">x =</span> <span class="st">"log2FoldChange"</span>,</span>
<span id="cb108-69"><a href="#cb108-69" aria-hidden="true" tabindex="-1"></a>                  <span class="at">y =</span> <span class="st">"padj"</span>,</span>
<span id="cb108-70"><a href="#cb108-70" aria-hidden="true" tabindex="-1"></a>                  <span class="co"># xlim = c(-3, 3),</span></span>
<span id="cb108-71"><a href="#cb108-71" aria-hidden="true" tabindex="-1"></a>                  <span class="co"># ylim = c(-0.5, 8),</span></span>
<span id="cb108-72"><a href="#cb108-72" aria-hidden="true" tabindex="-1"></a>                  <span class="at">title =</span> <span class="fu">paste</span>(con1, <span class="st">"vs"</span>, con2, <span class="at">sep =</span> <span class="st">" "</span>),</span>
<span id="cb108-73"><a href="#cb108-73" aria-hidden="true" tabindex="-1"></a>                  <span class="at">subtitle =</span> <span class="st">"DESeq2 Results"</span>,</span>
<span id="cb108-74"><a href="#cb108-74" aria-hidden="true" tabindex="-1"></a>                  <span class="co"># FCcutoff = ,</span></span>
<span id="cb108-75"><a href="#cb108-75" aria-hidden="true" tabindex="-1"></a>                  <span class="at">pCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb108-76"><a href="#cb108-76" aria-hidden="true" tabindex="-1"></a>                  <span class="co"># #labFace = "bold",</span></span>
<span id="cb108-77"><a href="#cb108-77" aria-hidden="true" tabindex="-1"></a>                  <span class="at">legendPosition =</span> <span class="st">"none"</span>,</span>
<span id="cb108-78"><a href="#cb108-78" aria-hidden="true" tabindex="-1"></a>                  <span class="at">caption =</span> <span class="st">""</span>) <span class="sc">|&gt;</span> </span>
<span id="cb108-79"><a href="#cb108-79" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="fu">paste0</span>(results.folder, name1, <span class="st">".VS."</span>, name2, <span class="st">".volcano.png"</span>))</span>
<span id="cb108-80"><a href="#cb108-80" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-81"><a href="#cb108-81" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Pathway summary plot</span></span>
<span id="cb108-82"><a href="#cb108-82" aria-hidden="true" tabindex="-1"></a>  p <span class="ot">&lt;-</span> <span class="fu">gs_summary_overview</span>(<span class="at">res_enrich =</span> subset.topgo, <span class="at">n_gs =</span> <span class="dv">15</span>) <span class="sc">+</span></span>
<span id="cb108-83"><a href="#cb108-83" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Pathway Enrichment"</span>,</span>
<span id="cb108-84"><a href="#cb108-84" aria-hidden="true" tabindex="-1"></a>         <span class="at">subtitle =</span> <span class="fu">paste</span>(con1, <span class="st">"vs"</span>, con2, <span class="at">sep =</span> <span class="st">" "</span>))</span>
<span id="cb108-85"><a href="#cb108-85" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb108-86"><a href="#cb108-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="fu">paste0</span>(results.folder, name1, <span class="st">".VS."</span>, name2, <span class="st">".pathways.png"</span>),</span>
<span id="cb108-87"><a href="#cb108-87" aria-hidden="true" tabindex="-1"></a>         <span class="at">plot =</span> p,</span>
<span id="cb108-88"><a href="#cb108-88" aria-hidden="true" tabindex="-1"></a>         <span class="at">width =</span> <span class="dv">11</span>,</span>
<span id="cb108-89"><a href="#cb108-89" aria-hidden="true" tabindex="-1"></a>         <span class="at">height =</span> <span class="dv">8</span>,</span>
<span id="cb108-90"><a href="#cb108-90" aria-hidden="true" tabindex="-1"></a>         <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb108-91"><a href="#cb108-91" aria-hidden="true" tabindex="-1"></a>         <span class="at">bg =</span> <span class="st">"white"</span>)</span>
<span id="cb108-92"><a href="#cb108-92" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="list-for-function" class="level2">
<h2 class="anchored" data-anchor-id="list-for-function">List for function</h2>
<p>Now we can create our list for the function input.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a>con1 <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Mixed CD4 HLA B"</span>, <span class="st">"Mixed CD4 HLA B"</span>, <span class="st">"Mixed CD4 HLA B"</span>, <span class="st">"Mixed CD8-HLA B"</span>, <span class="st">"Mixed CD8-HLA B"</span>, <span class="st">"Mixed CD8-HLA B"</span>, <span class="st">"Autol C57 CD4"</span>, <span class="st">"Autol C57 CD4"</span>, <span class="st">"Autol C57-CD8"</span>, <span class="st">"Autol C57-CD8"</span>, <span class="st">"Mixed CD4 HLA B"</span>, <span class="st">"Mixed CD8-HLA D"</span>)</span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a>con2 <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Reject CD4"</span>, <span class="st">"Autol C57 CD4"</span>, <span class="st">"WT control CD4"</span>, <span class="st">"Reject CD8"</span>, <span class="st">"Autol C57-CD8"</span>, <span class="st">"WT control CD8"</span>, <span class="st">"Reject CD4"</span>, <span class="st">"WT control CD4"</span>, <span class="st">"Reject CD8"</span>, <span class="st">"WT control CD8"</span>, <span class="st">"Reject CD4"</span>, <span class="st">"Autol Balb-CD8"</span>)</span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-5"><a href="#cb109-5" aria-hidden="true" tabindex="-1"></a>name1 <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"mix.4.b"</span>, <span class="st">"mix.4.b"</span>, <span class="st">"mix.4.b"</span>, <span class="st">"mix.8.b"</span>, <span class="st">"mix.8.b"</span>, <span class="st">"mix.8.b"</span>, <span class="st">"auto.4.b"</span>, <span class="st">"auto.4.b"</span>, <span class="st">"auto.8.b"</span>, <span class="st">"auto.8.b"</span>, <span class="st">"mix.4.d"</span>, <span class="st">"mix.8.d"</span>)</span>
<span id="cb109-6"><a href="#cb109-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-7"><a href="#cb109-7" aria-hidden="true" tabindex="-1"></a>name2 <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"rej.4.b"</span>, <span class="st">"auto.4.b"</span>, <span class="st">"wt.con.4.b"</span>, <span class="st">"rej.8.b"</span>, <span class="st">"auto.8.b"</span>, <span class="st">"wt.con.8.b"</span>, <span class="st">"rej.4.b"</span>, <span class="st">"wt.con.4.b"</span>, <span class="st">"rej.8.b"</span>, <span class="st">"wt.con.8.b"</span>, <span class="st">"auto.4.d"</span>, <span class="st">"auto.8.d"</span>)</span>
<span id="cb109-8"><a href="#cb109-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-9"><a href="#cb109-9" aria-hidden="true" tabindex="-1"></a>function.list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">list</span>(con1), <span class="fu">list</span>(con2), <span class="fu">list</span>(name1), <span class="fu">list</span>(name2))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="run-the-function" class="level2">
<h2 class="anchored" data-anchor-id="run-the-function">Run the function</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb110"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb110-1"><a href="#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pmap</span>(function.list, plot_results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(results.folder): '2024_01_19_results\mix.4.b.VS.rej.4.b'
already exists</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image
Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-1.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(results.folder): '2024_01_19_results\mix.4.b.VS.auto.4.b'
already exists</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-2.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-3.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(results.folder):
'2024_01_19_results\mix.4.b.VS.wt.con.4.b' already exists</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-4.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-5.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(results.folder): '2024_01_19_results\mix.8.b.VS.rej.8.b'
already exists</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-6.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-7.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(results.folder): '2024_01_19_results\mix.8.b.VS.auto.8.b'
already exists</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-8.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-9.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(results.folder):
'2024_01_19_results\mix.8.b.VS.wt.con.8.b' already exists</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-10.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-11.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(results.folder): '2024_01_19_results\auto.4.b.VS.rej.4.b'
already exists</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-12.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-13.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(results.folder):
'2024_01_19_results\auto.4.b.VS.wt.con.4.b' already exists</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-14.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-15.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(results.folder): '2024_01_19_results\auto.8.b.VS.rej.8.b'
already exists</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-16.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-17.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(results.folder):
'2024_01_19_results\auto.8.b.VS.wt.con.8.b' already exists</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-18.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-19.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: One or more p-values is 0. Converting to 10^-1 * current lowest
non-zero p-value...</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(results.folder): '2024_01_19_results\mix.4.d.VS.auto.4.d'
already exists</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-20.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-21.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dir.create(results.folder): '2024_01_19_results\mix.8.d.VS.auto.8.d'
already exists</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-22.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-23.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
<div class="cell-output-display">
<p><img src="2024_02_09_Code_Review_files/figure-html/unnamed-chunk-40-24.png" class="img-fluid" width="672"></p>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[[1]]
[1] "2024_01_19_results/mix.4.b.VS.rej.4.b/mix.4.b.VS.rej.4.b.pathways.png"

[[2]]
[1] "2024_01_19_results/mix.4.b.VS.auto.4.b/mix.4.b.VS.auto.4.b.pathways.png"

[[3]]
[1] "2024_01_19_results/mix.4.b.VS.wt.con.4.b/mix.4.b.VS.wt.con.4.b.pathways.png"

[[4]]
[1] "2024_01_19_results/mix.8.b.VS.rej.8.b/mix.8.b.VS.rej.8.b.pathways.png"

[[5]]
[1] "2024_01_19_results/mix.8.b.VS.auto.8.b/mix.8.b.VS.auto.8.b.pathways.png"

[[6]]
[1] "2024_01_19_results/mix.8.b.VS.wt.con.8.b/mix.8.b.VS.wt.con.8.b.pathways.png"

[[7]]
[1] "2024_01_19_results/auto.4.b.VS.rej.4.b/auto.4.b.VS.rej.4.b.pathways.png"

[[8]]
[1] "2024_01_19_results/auto.4.b.VS.wt.con.4.b/auto.4.b.VS.wt.con.4.b.pathways.png"

[[9]]
[1] "2024_01_19_results/auto.8.b.VS.rej.8.b/auto.8.b.VS.rej.8.b.pathways.png"

[[10]]
[1] "2024_01_19_results/auto.8.b.VS.wt.con.8.b/auto.8.b.VS.wt.con.8.b.pathways.png"

[[11]]
[1] "2024_01_19_results/mix.4.d.VS.auto.4.d/mix.4.d.VS.auto.4.d.pathways.png"

[[12]]
[1] "2024_01_19_results/mix.8.d.VS.auto.8.d/mix.8.d.VS.auto.8.d.pathways.png"</code></pre>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>